tmva/tmva/src/RBDT.cxx

```diff
@@ -1,4 +1,376 @@
-#include "TMVA/RBDT.hxx"
+/**********************************************************************************
+ * Project: ROOT - a Root-integrated toolkit for multivariate data analysis       *
+ * Package: TMVA                                                                  *
+ *                                                                                *
+ *                                                                                *
+ * Description:                                                                   *
+ *                                                                                *
+ * Authors:                                                                       *
+ *      Jonas Rembser (jonas.rembser@cern.ch)                                     *
+ *                                                                                *
+ * Copyright (c) 2024:                                                            *
+ *      CERN, Switzerland                                                         *
+ *                                                                                *
+ * Redistribution and use in source and binary forms, with or without             *
+ * modification, are permitted according to the terms listed in LICENSE           *
+ * (see tmva/doc/LICENSE)                                          *
+ **********************************************************************************/
 
-template class TMVA::Experimental::RBDT<TMVA::Experimental::BranchlessForest<float>>;
-template class TMVA::Experimental::RBDT<TMVA::Experimental::BranchlessJittedForest<float>>;
+#include <TMVA/RBDT.hxx>
+
+#include <ROOT/RSpan.hxx>
+#include <ROOT/StringUtils.hxx>
+
+#include <TFile.h>
+#include <TSystem.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <stdlib.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace {
+
+namespace detail {
+
+typedef std::unordered_map<int, int> IndexMap;
```

Comment:
What is the meaning of the keys and values in this map type?

Reply:
I have added some docs now, but I think it's not necessary to go into detail here.

In the future, I think we can get rid of it anyway: at some point the XGBoost devs took inspiration from my library to optimize their own BDT representation, if I remember the discussions on GitHub correctly.

So I think we can avoid this index translation in the future by parsing XGBoosts internal representation directly instead of going over the `txt` dump.